packages/ses-ava/src/types.js

```diff
@@ -42,13 +42,16 @@
  */
 
 /**
- * @callback ImplFunc
+ * @callback ImplFuncFunc
```

Comment:
I don't understand your changes to the types. Let's discuss.

Reply:
The issue is the Ava API design, which I will make no attempt to defend.  Basically, the macro feature has you attach a name generator function to the test implementation function by assigning it to a property named `title`, so you need to define a type that describes something that is both a function taking an `Assertions` parameter and an object that can have a `title` property whose value is a function returning a string.  The type-fu was worked out with @kriskowal, since my understanding of TypeScript-in-doc-comments is somewhere between poor and terrible.  If you prefer a different way to define this that is more elegant / more cutting with the grain / more pleasing to your own aesthetic sensibilities, then I'm all for it, 'cause I think this one is dreadful.  But it may be the price of compatibility with an API designed by drunken squirrels.